homeassistant/components/switcher_kis/__init__.py

```diff
@@ -78,21 +74,6 @@
 )
 
 
-@bind_hass
-async def _validate_edit_permission(
-    hass: HomeAssistantType, context: ContextType, entity_id: str
-) -> None:
-    """Use for validating user control permissions."""
-    splited = split_entity_id(entity_id)
-    if splited[0] != SWITCH_DOMAIN or not splited[1].startswith(DOMAIN):
```

Comment:
Although looking at the forum post log, the problem seems to be that the user is unknown.

Reply:
First of all, thanks a lot for your time.

Another problem is that the integration is implemented in such way that it supports only one device
and if you look in the actual `turn_on_with_timer` service's code you see that there is no reference to the `entity_id` while calling to the device's API, actually this is the same for both services, you can see it in `__init__.py` under `async_set_auto_off_service` and `async_turn_on_with_timer_service`:

`async with SwitcherV2Api( hass.loop, device_data.ip_addr, phone_id, device_id, device_password ) as swapi:`
`await swapi.control_device(COMMAND_ON, service.data[CONF_TIMER_MINUTES])`

So, even if I will use entity service, as long as the integration supports only one device and the entity_id is not taken into consideration while calling the device API - In case I will call the service with my `switch.room_light` entity_id it will turn on the boiler, regardless what entity I sent to the service.

So I need some way to enable the service only for `switcher` devices (same as we cannot call `light.turn_on` on a `switch` entity)
And as long as there is no support for multiple devices and the service implementation will turn on the same device regardless what entity_id was sent - I think we can remove the `entity_id` var from the service.

Feel free to correct me if I wrong :)

Reply:
Ok I will give it a try